File: Selenium/Activities/selenium_Activity4.java
package selenium_session1;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.Color;

public class selenium_Activity4 
{
	public static void main(String[] arg)
	{
		//create a WebDriver instance
		WebDriver driver = new FirefoxDriver();
				
		//go to a web page
		driver.get("https://training-support.net/webelements/target-practice");
		
		//do stuff on the page
		System.out.println("Title of the page is : " + driver.getTitle());
		
		//Find the 3rd header and print its text
        //String thirdHeaderText = driver.findElement(By.xpath("//h3[contains(text(), '#3')]")).getText();
        //System.out.println(thirdHeaderText);
        
		WebElement element = driver.findElement(By.xpath("/html/body/div/main/div/div/div/div/div[2]/h3[1]"));
		System.out.println("Text on 3rd header is: " + element.getText());
		
		//Find the 5th header and print its color
        Color fifthHeaderColor = Color.fromString(driver.findElement(By.xpath("//h5[contains(text(), '#5')]")).getCssValue("color"));
        System.out.println("Color as RGB of 5th header is: " + fifthHeaderColor.asRgb());
        System.out.println("Color as hexcode of 5th header is: " + fifthHeaderColor.asHex());
        
		element = driver.findElement(By.xpath("/html/body/div/main/div/div/div/div/div[2]/h5"));
		System.out.println("Color of 5th header is: " + element.getCssValue("color"));

        //Find the violet button and print its classes
        String purpleButtonClass = driver.findElement(By.xpath("/html/body/div/main/div/div/div/div/div[2]/button[5]")).getDomAttribute("class");
        System.out.println("classes of violet button is : " + purpleButtonClass);
        
        //Find the grey button and print its text
        System.out.println("text of grey button is : " + driver.findElement(By.xpath("//button[contains(@class, 'slate')]")).getText());

        
		//close the browser
		driver.quit();
	
	}
}

File: Selenium/Activities/selenium_activity4.py
from selenium import webdriver
from selenium.webdriver.common.by import By

#create WebDriver instance
with webdriver.Firefox() as driver:

    #go to web page
    driver.get("https://training-support.net/webelements/target-practice")

    #do the stuff on the page
    pageTitle = driver.title
    print("Title of the page is : " + pageTitle)

    #Find the 3rd header and print its text
    textHeader = driver.find_element(By.XPATH, "/html/body/div/main/div/div/div/div/div[2]/h3[1]").text
    print("Text on 3rd header is: " + textHeader)

    #Find the 5th header and print its color
    colorHeader = driver.find_element(By.XPATH, "//h5[contains(text(), '#5')]").value_of_css_property("color")
    print("color of 5th header is: " + colorHeader)
	
	#Find the violet button and print its classes
    classButton = driver.find_element(By.XPATH, "/html/body/div/main/div/div/div/div/div[2]/button[5]").get_attribute("class")
    print("classes of violet button is: " + classButton)

    #Find the grey button and print its text
    textHeader = driver.find_element(By.XPATH, "//button[contains(@class, 'slate')]").text
    print("Text on grey button is: " + textHeader)
    
    #close the browser
    driver.quit()